lib/membrane/video_compositor/scene/object.ex

```diff
@@ -0,0 +1,38 @@
+defmodule Membrane.VideoCompositor.Scene.Object do
+  @moduledoc """
+  This module holds common types for different kinds of Objects available.
+  """
+
+  alias Membrane.Pad
+  alias Membrane.VideoCompositor.Scene.{Layout, Resolution, Texture}
+
+  @typedoc """
+  Objects are renderable entities in VC, that can serve as input for other
+  objects or as an output of the video.
+  They are either Texture structs or structs defining Layouts
+  following Layout.t() definition.
+  """
+  @type t :: Layout.t() | Texture.t()
+
+  @typedoc """
+  Defines how an object can be referenced in Scene.
+  Objects can be assigned to names and identified
+  at other objects as inputs based on assigned names
+  """
+  @type name :: tuple() | atom()
+
+  @typedoc """
+  Defines how the input of an object can be specified
+  in Video Compositor.
+  """
+  @type input :: name() | Pad.name_t()
+
+  @typedoc """
+  Define how the output resolution of an object can be specified.
+  Additionally, in Textures resolution can be specified as
```

Comment:
How about defining an output_resolution for Textures and a separate output_resolution for Layouts, and simply defining `Object.object_output_resolution` as an alternative of these two types?
I believe it would help in separating the "types of resolutions" applicable to different object - the information that "...the Textures resolution can be specified as as a transformed resolution of the object" could be then available in the Texture module documentation.